File: number_of_subarrays_with_odd_sum.py
class Solution:
    def numOfSubArrays(self, arr: List[int]) -> int:
        current = 0
        odd = 0
        even = 0
        result = 0
        MOD = 10**9 + 7
        
        for n in arr:
            current += n
            
            if current % 2 != 0:
                result = (result + 1 + even) % MOD
                odd += 1
            else:
                result = (result + odd) % MOD
                even += 1
                
        return result

File: numberOfSubArraysWithOddSum.ts
function numOfSubArrays(arr: number[]): number{
    let current: number = 0
    let result: number = 0
    let even: number = 0
    let odd: number = 0

    const mod: number = 10**9 + 7

    for(let i = 0; i < arr.length; i++){
        current += arr[i]
        if(current % 2 !== 0){
            result = (result + 1 + even) % mod
            odd += 1
        }else{
            result = (result + odd) % mod
            even += 1
        }
    }
    return result
}

